src/main/java/org/openstreetmap/atlas/proto/adapters/ProtoIntegerArrayOfArraysAdapter.java

```diff
@@ -0,0 +1,64 @@
+package org.openstreetmap.atlas.proto.adapters;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.proto.ProtoIntegerArrayOfArrays;
+import org.openstreetmap.atlas.proto.ProtoSerializable;
+import org.openstreetmap.atlas.proto.converters.ProtoIntegerArrayOfArraysConverter;
+import org.openstreetmap.atlas.utilities.arrays.IntegerArrayOfArrays;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+
+/**
+ * Implements the {@link ProtoAdapter} interface to connect {@link IntegerArrayOfArrays} and
+ * {@link ProtoIntegerArrayOfArrays}.
+ *
+ * @author lcram
+ */
+public class ProtoIntegerArrayOfArraysAdapter implements ProtoAdapter
+{
+    private static final ProtoIntegerArrayOfArraysConverter CONVERTER = new ProtoIntegerArrayOfArraysConverter();
+
+    @Override
+    public ProtoSerializable deserialize(final byte[] byteArray)
+    {
+        ProtoIntegerArrayOfArrays protoIntegerArrayOfArrays = null;
+        try
+        {
+            protoIntegerArrayOfArrays = ProtoIntegerArrayOfArrays.parseFrom(byteArray);
+        }
+        catch (final InvalidProtocolBufferException exception)
+        {
+            throw new CoreException("Error encountered while parsing protobuf bytestream",
+                    exception);
+        }
+
+        final IntegerArrayOfArrays integerArrayOfArrays = CONVERTER
+                .convert(protoIntegerArrayOfArrays);
+
+        return integerArrayOfArrays;
+    }
+
+    @Override
+    public byte[] serialize(final ProtoSerializable serializable)
+    {
+        if (!(serializable instanceof IntegerArrayOfArrays))
+        {
+            throw new CoreException(
+                    "Invalid ProtoSerializable type was provided to {}: cannot serialize {}",
+                    this.getClass().getName(), serializable.getClass().getName());
+        }
+        final IntegerArrayOfArrays integerArrayOfArrays = (IntegerArrayOfArrays) serializable;
+
+        if (integerArrayOfArrays.size() > Integer.MAX_VALUE)
+        {
+            // TODO see note about this check in ProtoLongArrayOfArraysAdapter
```

Comment:
Maybe add this constraint in a single place (possibly a README on this) and remove the TODOs instances.

Reply:
After talking with Matthieu a bit about the size constraints, I think it might be best to just remove these checks. The arrays we pass in will never be that large anyway, and I was not consistent about enforcing that check across all the different adapters. If it becomes a problem, in the future I can take some time to implement that check correctly everywhere.